samples/kubernetes/end2end/dashboard/exporter-config.yaml

```diff
@@ -0,0 +1,57 @@
+host: localhost
+port: 7001
+metricsNameSnakeCase: true
+queries:
```

Comment:
due current changes extra element has to be added:
restPort: 8001
before the `queries` line

Reply:
Per the doc `restPort | Optional. Overrides the port on which the exporter should contact the REST API . Needed when behind a load balancer.`, it's only needed when behind a load balancer while this sample doesn't involved any load balancer. So don't need to specify the param `restPort`.

Reply:
All of our samples in  Kubernetes operator project and all customer use cases will have a load balancer.  Lily maybe you should consider adding a LB to this sample.

Reply:
Scraping through load balancers can not work correctly since you can hit a different instance on each scrape, which could collapse the dashboard. 

For example, in the Weblogic dashboard, we display the number of running servers and we display server overview, one row for every server. If scraping through a load balancer, the number of running servers might be displayed as 1. The server overview might display only one server at a time and rotate among the running servers. 
The actual behavior can be varied which depends on how we do query to select, group and aggregate metrics in the dashboard. But hope you get the point.

Reply:
Just FYI...In our current setup, we have the LB configured and in the exporter-config.yaml we are using the restPort to specify
host: 132.145.16.40
port: 30305
metricsNameSnakeCase: true
restPort: 8001
queries:
- key: name
  keyName: location
  prefix: wls_server_
  applicationRuntimes:
    key: name
    keyName: app
    componentRuntimes:
      prefix: wls_webapp_config_
      type: WebAppComponentRuntime
      key: name
      values: [deploymentState, contextRoot, sourceInfo, sessionsOpenedTotalCount, openSessionsCurrentCount, openSessionsHighCount]
      servlets:
        prefix: wls_servlet_
        key: servletName

With this settings also,we are able to scrape the WLS metrics and display in the Weblogic dashboard. When we have more than 1 server configured, it displays the details for all servers.

Reply:
How you configure Prometheus? Follow the steps in the sample? 
How many targets are shown in the Premetheus target page(via http://<host>:30000/targets)?

Reply:
If you follow the steps to configure Prometheus, Premetheus server will create one target for every running managed server. That means  Premetheus server will scrape metrics from every server directly, not from load balancer. I don't think you need to configure `restPort` in exporter-config.yaml case.

Reply:
I agree. The restPort feature was added to deal with a specific integration test that accessed the metrics via a load-balancer, but that is not a common use case. It is pretty much only there to allow manual inspection of the metrics.